cluster/replication/shard_replication_engine.go

```diff
@@ -104,7 +108,10 @@ func (s *shardReplicationEngine) startShardReplication(op shardReplicationOp) {
 
 	go func() {
 		defer s.ongoingReplications.Add(-1)
-
+		// TODO defer deleting the op from ongoing ops map and fsm maps as well? but only if it doesn't work?
```

Comment:
Planning on coming back to this later after this PR is merged, leaving the TODO so I don't forget